lib/engine/share_price.rb

```diff
@@ -2,51 +2,56 @@
 
 module Engine
   class SharePrice
-    attr_reader :coordinates, :price, :color, :corporations, :can_par, :end_game_trigger
+    attr_reader :coordinates, :price, :color, :corporations, :can_par, :end_game_trigger,
+                :type
 
     def self.from_code(code, row, column, unlimited_colors, multiple_buy_colors: [])
       return nil if !code || code == ''
 
       price = code.scan(/\d/).join('').to_i
-      can_par = code.include?('p')
-      end_game_trigger = code.include?('e')
-      color =
+
+      color, type =
         case
-        when can_par
-          :red
-        when end_game_trigger
-          :blue
+        when code.include?('p')
+          %i[red par]
+        when code.include?('e')
+          %i[blue endgame]
         when code.include?('blk')
-          :black
+          %i[black close]
         when code.include?('b')
-          :brown
+          %i[brown multiple_buy]
         when code.include?('o')
-          :orange
+          %i[orange unlimited]
         when code.include?('y')
-          :yellow
+          %i[yellow no_cert_limit]
+        when code.include?('l')
+          %i[red liquidation]
+        when code.include?('a')
+          %i[gray acquisition]
         end
 
       SharePrice.new([row, column],
                      price: price,
-                     can_par: can_par,
-                     end_game_trigger: end_game_trigger,
                      color: color,
+                     type: type,
                      unlimited_colors: unlimited_colors,
                      multiple_buy_colors: multiple_buy_colors)
     end
 
     def initialize(coordinates,
                    price:,
-                   can_par: false,
-                   end_game_trigger: false,
                    color: nil,
+                   type: nil,
                    unlimited_colors: [],
                    multiple_buy_colors: [])
       @coordinates = coordinates
       @price = price
       @color = color
-      @can_par = can_par
-      @end_game_trigger = end_game_trigger
+      @type = type
+      @can_par = type == :par
+      @end_game_trigger = type == :endgame
```

Comment:
do we need vars for all of these? can we just have @type?

Reply:
or perhaps a stock market space could have multiple types, not sure if that's needed now

Reply:
I can mod the existing ones to not need the variables.